### PR TITLE
Added pages to view and edit admin profile.

### LINKED ISSUE #554: Add edit/view profile option for the admin
There is option for the volunteer to view its profile as well as edit it. For the admin, no such option is available.

## Mocks
vms_issue4
vms_issue42

## Acceptance Criteria
Admin user should be able to view profile as well as edit is just like the volunteers.

### PR DESCRIPTION
**Pull request by sammy1997**
# Description
Added a service to get admin object by id. Added profile view and edit templates and views. Updated base.html and urls.py accordingly

Fixes #554 

# Type of Change:
- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
profile
xyz
xyzz

# Checklist:

- [ ] My PR follows the style guidelines of this project
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] Any dependent changes have been merged 
- [ ] My changes generate no new warnings 
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by sammy1997** [comment]
@tapasweni-pathak @tonythomas01 Please review this PR. Thanks.

File: vms/administrator/services.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+def get_administrator_by_id(admin_id):
+
+    is_valid = True
+    result = None
+
+    try:
+        administrator = Administrator.objects.get(pk=admin_id)
+    except ObjectDoesNotExist:
```
Comment (tonythomas01) [inline vms/administrator/services.py]: `except Administrator.ObjectDoesNotExist:`

File: vms/administrator/services.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from django.core.exceptions import ObjectDoesNotExist
+from administrator.models import Administrator
+
+
+def get_administrator_by_id(admin_id):
```
Comment (tonythomas01) [inline vms/administrator/services.py]: Why do you want to do all this, when you can just: 
`return  Administrator.objects.filter(pk=admin_id).first() `

File: vms/administrator/views.py [hunk 3]
```diff
@@ -102,3 +103,43 @@ def post(self, request, *args, **kwargs):
 @admin_required
 def settings(request):
     return HttpResponseRedirect(reverse('event:list'))
+
+
+# View admin profile
+@login_required
+@admin_required
+def profile(request, admin_id):
```
Comment (tonythomas01) [inline vms/administrator/views.py]: please use CBV instead.

File: vms/administrator/views.py [hunk 4]
```diff
@@ -102,3 +103,43 @@ def post(self, request, *args, **kwargs):
...
+
+@login_required
+@admin_required
+def edit_profile(request, admin_id):
+    admin = get_administrator_by_id(admin_id)
+    organization_list = get_organizations_ordered_by_name()
+    if not admin:
+        return render(request, 'administrator/404.html')
+    else:
+        print request.POST
```
Comment (tonythomas01) [inline vms/administrator/views.py]: why is this here ?

File: vms/administrator/services.py (hunk 2 above)
Comment (sammy1997) [reply vms/administrator/services.py]: @tonythomas01 I read the current code, and saw that this was the convention that was being followed. Like in volunteer/ services.py . Should I delete this file then?

File: vms/administrator/services.py (hunk 1 above)
Comment (sammy1997) [reply vms/administrator/services.py]: @tonythomas01 Same has been done in get_volunteer_by_id in volunteer/services.py . Do I change this then?

File: vms/administrator/views.py (hunk 4 above)
Comment (sammy1997) [reply vms/administrator/views.py]: I had it for debugging. I will change this. Thanks.

**Comment by sammy1997** [comment]
@tonythomas01 @tapasweni-pathak Please review this PR again. I have made the required changes.

File: vms/administrator/views.py [hunk 5]
```diff
@@ -102,3 +103,66 @@ def post(self, request, *args, **kwargs):
 @admin_required
 def settings(request):
     return HttpResponseRedirect(reverse('event:list'))
+
+
+'''
+    The View to edit Admin Profile
```
[the remark this answers is not in the document]
Comment (sammy1997) [reply vms/administrator/views.py]: It's a comment @codacy-bot

**Comment by sammy1997** [comment]
@MeepyMay I think some of the comments by Codacy bot are not meaningful. Can you please delete these ? @tonythomas01 @tapasweni-pathak Please review this PR. I have made the required changes as suggested by @tonythomas01

File: vms/administrator/utils.py [hunk 6]
```diff
@@ -14,3 +16,23 @@ def wrapped_view(request, *args, **kwargs):
...
+
+
+def admin_id_check(func):
+    @wraps(func)
+    def wrapped_view(request, admin_id):
+        administrator = getattr(request.user, 'administrator',
+                      hasattr(request.user, 'administrator'))
+        if administrator is None:
+            return render(request, 'vms/no_admin_rights.html', status=403)
+        elif administrator is False:
```
Comment (tonythomas01) [inline vms/administrator/utils.py]: if not administrator

File: vms/administrator/utils.py [hunk 7]
```diff
@@ -14,3 +16,23 @@ def wrapped_view(request, *args, **kwargs):
...
 
     return wrapped_view
+
+
+def admin_id_check(func):
+    @wraps(func)
+    def wrapped_view(request, admin_id):
+        administrator = getattr(request.user, 'administrator',
+                      hasattr(request.user, 'administrator'))
+        if administrator is None:
```
Comment (tonythomas01) [inline vms/administrator/utils.py]: what if administrator is set ?

File: vms/administrator/utils.py [hunk 8]
```diff
@@ -14,3 +16,23 @@ def wrapped_view(request, *args, **kwargs):
...
+            return render(request, 'vms/no_admin_rights.html', status=403)
+        elif administrator is False:
+            admin = Administrator.objects.get(id=admin_id)
+            if not admin:
```
Comment (tonythomas01) [inline vms/administrator/utils.py]: this will never happen. If an admin cannot be found on a `get`, it would raise an Administrator.DoesNotExist

File: vms/administrator/utils.py (hunk 7 above)
Comment (sammy1997) [reply vms/administrator/utils.py]: Can you please explain what you mean? @tonythomas01

File: vms/administrator/utils.py (hunk 8 above)
Comment (sammy1997) [reply vms/administrator/utils.py]: Ok. So i will add a try except then? @tonythomas01

File: vms/administrator/utils.py (hunk 7 above)
Comment (sammy1997) [reply vms/administrator/utils.py]: @tonythomas01 Please explain what you mean.

File: vms/administrator/utils.py (hunk 8 above)
Comment (tonythomas01) [reply vms/administrator/utils.py]: no. Line 30 will not be executed by any chance. The whole function needs to be improved programmatically.

**Comment by sammy1997** [comment]
@tonythomas01 @MeepyMay @tapasweni-pathak I have changed the commit for util as suggested by @tonythomas01 . Please review the PR again.

**Comment by sammy1997** [comment]
@MeepyMay @tonythomas01 Please review this

**Comment by tapasweni-pathak** [comment]
Ty for the work! 🎉
